packages/meditrak-server/src/routes/exportSurveyResponses.js

```diff
@@ -83,76 +84,43 @@ export async function exportSurveyResponses(req, res) {
   const workbook = { SheetNames: [], Sheets: {} };
 
   try {
+    const variablesExtractor = new SurveyResponseVariablesExtractor(models);
     let country;
     let entities;
-    let reportName;
-    if (viewId) {
-      const dashboardReport = await models.dashboardReport.findById(viewId);
-      reportName = dashboardReport.viewJson.name;
-    }
+    const reportName = viewId && (await models.dashboardReport.findById(viewId).viewJson.name);
+
     if (countryCode) {
-      country = await models.country.findOne({ code: countryCode });
-      countryId = country.id;
+      const variables = await variablesExtractor.getVariablesByCountryCode(countryCode);
+      country = variables.country;
+      countryId = variables.countryId;
     }
+
     if (entityCode) {
-      const entity = await models.entity.findOne({ code: entityCode });
-      country = await models.country.findOne({ code: entity.country_code });
-      entities = [entity];
+      const variables = await variablesExtractor.getVariablesByEntityCode(entityCode);
+      country = variables.country;
+      entities = variables.entities;
     } else if (countryId) {
-      country = await models.country.findById(countryId);
-      entities = await models.entity.find({ country_code: country.code }, { sort: ['name ASC'] });
+      const variables = await variablesExtractor.getVariablesByCountryId(countryId);
+      country = variables.country;
+      entities = variables.entities;
     } else if (entityIds) {
-      entities = await Promise.all(
-        entityIds.split(',').map(entityId => models.entity.findById(entityId)),
-      );
-      if (!countryId && entities.length > 0) {
-        const countryCodeFromEntity = entities[0].country_code;
-        country = await models.country.findOne({ code: countryCodeFromEntity });
-        countryId = country.id;
-      }
+      const variables = await variablesExtractor.getVariablesByEntityIds(entityIds);
+      country = variables.country;
+      countryId = variables.countryId;
+      entities = variables.entities;
```

Comment:
could you refactor this further to:
```
const getParametersFromInput = (variablesExtractor, a, b, ...) => {
 if (a) return variablesExtractor.getVariablesByCountryId(a);
...
else{ throw }
}
```
and replace the logic here with:
```javascript
const { country, entities, ... } = await getParametersFromInput(variablesExtractor, entitityCode, countryId, ...);
```

Reply:
Sure! This looks better.